lib/canary/plugs.ex

```diff
@@ -193,11 +193,12 @@ defmodule Canary.Plugs do
     current_user = Dict.fetch! conn.assigns, current_user_name
     action = get_action(conn)
     is_persisted = persisted?(opts)
+    non_id_actions = if opts[:non_id], do: ([:index, :new, :create] ++ opts[:non_id]), else: [:index, :new, :create]
```

Comment:
the `++` operator will create an improper list if a non-list is passed as the second argument, e.g. `[:show] ++ :derp` gives `[:show | :derp]`, which will silently pass through `in` unless we look for a non-existant member or the member on the right side of the `|`. 

I think a better way would be to force users to pass a list of actions, and use `Enum.concat([:index, :new, :create], opts[:non_id])`, which will `raise` if the second argument is not a list.